ibtk/src/lagrangian/JacobianCalculator.cpp

```diff
@@ -0,0 +1,509 @@
+    const double Jac_21 = p2(2) - p0(2);
+    const double Jac_22 = p3(2) - p0(2);
+
+    const double J = Jac_00 * (Jac_11 * Jac_22 - Jac_12 * Jac_21) - Jac_10 * (Jac_01 * Jac_22 - Jac_02 * Jac_21) +
+    Jac_20 * (Jac_01 * Jac_12 - Jac_02 * Jac_11);
+
+    TBOX_ASSERT(J > 0.0);
+    for (double& jxw : d_JxW) jxw *= J;
+
+    return d_JxW;
+}
+
+const std::vector<double>&
+Hex8JacobianCalculator::get_JxW(const Elem* elem)
+{
+    TBOX_ASSERT(elem->type() == std::get<0>(d_quad_key));
+    std::copy(d_quad_weights.begin(), d_quad_weights.end(), d_JxW.begin());
+#if 0
+    // calculate constants in Jacobians here
+    const Point p0 = elem->point(0);
+    const Point p1 = elem->point(1);
+    const Point p2 = elem->point(2);
+    const Point p3 = elem->point(3);
+    const Point p4 = elem->point(4);
+    const Point p5 = elem->point(5);
+    const Point p6 = elem->point(6);
+    const Point p7 = elem->point(7);
+
+    const double a_00 = 0.125 * (-p0(0) + p1(0) + p2(0) - p3(0) - p4(0) + p5(0) + p6(0) - p7(0));
+    const double b_00 = 0.125 * (p0(0) - p1(0) + p2(0) - p3(0) + p4(0) - p5(0) + p6(0) - p7(0));
+    const double c_00 = 0.125 * (p0(0) - p1(0) - p2(0) + p3(0) - p4(0) + p5(0) + p6(0) - p7(0));
+    const double d_00 = 0.125 * (-p0(0) + p1(0) - p2(0) + p3(0) + p4(0) - p5(0) + p6(0) - p7(0));
+    const double a_10 = 0.125 * (-p0(1) + p1(1) + p2(1) - p3(1) - p4(1) + p5(1) + p6(1) - p7(1));
+    const double b_10 = 0.125 * (p0(1) - p1(1) + p2(1) - p3(1) + p4(1) - p5(1) + p6(1) - p7(1));
+    const double c_10 = 0.125 * (p0(1) - p1(1) - p2(1) + p3(1) - p4(1) + p5(1) + p6(1) - p7(1));
+    const double d_10 = 0.125 * (-p0(1) + p1(1) - p2(1) + p3(1) + p4(1) - p5(1) + p6(1) - p7(1));
+    const double a_20 = 0.125 * (-p0(2) + p1(2) + p2(2) - p3(2) - p4(2) + p5(2) + p6(2) - p7(2));
+    const double b_20 = 0.125 * (p0(2) - p1(2) + p2(2) - p3(2) + p4(2) - p5(2) + p6(2) - p7(2));
+    const double c_20 = 0.125 * (p0(2) - p1(2) - p2(2) + p3(2) - p4(2) + p5(2) + p6(2) - p7(2));
+    const double d_20 = 0.125 * (-p0(2) + p1(2) - p2(2) + p3(2) + p4(2) - p5(2) + p6(2) - p7(2));
+
+    for (unsigned int i = 0; i < d_JxW.size(); i++)
+    {
+        // calculate Jacobians here
+        const double x = d_quad_points[i](0);
+        const double y = d_quad_points[i](1);
+        const double z = d_quad_points[i](2);
+        const double Jac_00 = 0;//a_1 + c_1 * y;
+        const double Jac_01 = 0;//b_1 + c_1 * x;
+        const double Jac_02 = 0;
+        const double Jac_10 = 0;//a_2 + c_2 * y;
+        const double Jac_11 = 0;//b_2 + c_2 * x;
+        const double Jac_12 = 0;
+        const double Jac_20 = 0;
+        const double Jac_21 = 0;
+        const double Jac_22 = 0;
+
+        const double J = Jac_00 * (Jac_11 * Jac_22 - Jac_12 * Jac_21) - Jac_10 * (Jac_01 * Jac_22 - Jac_02 * Jac_21) +
+        Jac_20 * (Jac_01 * Jac_12 - Jac_02 * Jac_11);
+
+        TBOX_ASSERT(J > 0.0);
+        d_JxW[i] *= J;
+    }
+#endif
+    return d_JxW;
+}
```

Comment:
Lets just delete this class for now - we can always go back and finish it later.